src/pathfinder/water_regions.cpp

```diff
@@ -66,6 +70,14 @@ class WaterRegion
 		return (TileX(tile) - TileX(this->tile_area.tile)) + WATER_REGION_EDGE_LENGTH * (TileY(tile) - TileY(this->tile_area.tile));
 	}
 
+	inline bool HasNonMatchingPatchLabel(TWaterRegionPatchLabel expected_label) const
+	{
+		for (TWaterRegionPatchLabel label : *this->tile_patch_labels) {
+			if (label != expected_label) return true;
+		}
+		return false;
+	}
```

Comment:
Alternatively you could keep a counter that keeps track of how many times label 1 is assigned during ForceUpdate. Then you can simply check if that counter is equal to the number of region tiles. That would save another iteration over the entire label array, and the need for this function :)

Oh and if you decide to keep it the way it is, document this function.

Reply:
Iterating the array that has just been written is very cheap. This seems simpler/more fool-proof than special-casing label 1 in the fill process. I'll add some suitable documentation later.